### PR TITLE
Issue 16462 create tag suggestions endpoint

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/TagResource.java [hunk 1]
```diff
@@ -44,16 +52,52 @@ protected TagResource(TagAPI tagAPI, WebResource webResource) {
...
+    public Map<String, RestTag> list(@Context final HttpServletRequest request,
+            @QueryParam("name") final String  tagName,
+            @QueryParam("siteOrFolder") final String siteOrFolderId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+
+        final User user = initDataObject.getUser();
+
+        List<Tag> tags;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: you might reduce this to
```
final List<Tag> tags = UtilMethods.isSet(tagName)?
    searchTagsInternal(tagName, siteOrFolderId, user): getTagsInternal();
```

File: dotCMS/src/main/java/com/dotcms/rest/TagResource.java [hunk 2]
```diff
@@ -44,16 +52,52 @@ protected TagResource(TagAPI tagAPI, WebResource webResource) {
...
+
+        if(UtilMethods.isSet(tagName)) {
+            tags = searchTagsInternal(tagName, siteOrFolderId, user);
+        } else {
+            tags = getTagsInternal();
+        }
+
         Map<String, RestTag> hash = Maps.newHashMapWithExpectedSize(tags.size());
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: set to final 
sane for transform and for(final Tag tag

File: dotCMS/src/main/java/com/dotcms/rest/TagResource.java [hunk 3]
```diff
@@ -44,16 +52,52 @@ protected TagResource(TagAPI tagAPI, WebResource webResource) {
...
+        TagTransform transform = new TagTransform();
         for (Tag tag : tags) {
             hash.put(tag.getTagName(), transform.appToRest(tag));
         }
         return hash;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: this is not following the standard response used for the endpoint ResponseEntityView

File: dotCMS/src/main/java/com/dotcms/rest/TagResource.java [hunk 4]
```diff
@@ -44,16 +52,52 @@ protected TagResource(TagAPI tagAPI, WebResource webResource) {
...
     }
 
+    private List<Tag> searchTagsInternal(final String tagName, final String siteOrFolderId,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: You can create TagHelper and put this there

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/TagResource.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: Talked to will about it and he doesn't want to change which means creating a new version of this

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/TagResource.java]: Already discussed @jdotcms

**Pull request merged**